simple-dns/src/dns/character_string.rs

```diff
@@ -25,8 +25,10 @@ impl<'a> CharacterString<'a> {
         if data.len() > MAX_CHARACTER_STRING_LENGTH {
             return Err(SimpleDnsError::InvalidCharacterString);
         }
-
-        Ok(Self { data })
+        match String::from_utf8(data.clone().into_owned()) {
```

Comment:
Check for valid utf-8 (and add a test for it), to be able to unwrap at line 44 safely.

Reply:
I think this check can't be used here, because a character string is not necessarily valid utf-8. If you check [this](https://datatracker.ietf.org/doc/html/rfc1035#section-3.3) section of the RFC-1035, it says that character string must be treated as binary data.

Since this library can receive records from other libraries, the safest place to check for a valid utf8 string is in the ''Into String" function, because it wouldn't impose valid utf8 when using the library to interface with something else.